### PR TITLE
Fix checkbox demos

### LINKED ISSUE #117: Improve vaadin-checkbox-flow demos

### PR DESCRIPTION
**Pull request by shadikhani**
Fix #117

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-checkbox-flow/120)

### CODE REVIEW

File: vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java [hunk 1]
```diff
@@ -34,174 +47,223 @@
...
     }
 
-    private void addReadOnlyGroup() {
+    private void indeterminateCheckbox() {
         // begin-source-example
-        // source-example-heading: Read-only checkbox group
-        Div valueInfo = new Div();
+        // source-example-heading: Indeterminate checkbox
+        Checkbox checkbox = new Checkbox("Select all");
+        checkbox.setIndeterminate(true);
```
Comment (tomivirkki) [inline vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java]: Why is this initially indeterminate even though neither of the checkboxes is selected?

File: vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java [hunk 2]
```diff
@@ -34,174 +47,223 @@
...
+        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
+        Set<String> items = new LinkedHashSet<>(
+                Arrays.asList("Option one", "Option two"));
+        checkboxGroup.setItems(items);
+        List<CheckboxGroup> checkboxList = new ArrayList<>();
+        checkboxList.add(checkboxGroup);
+        checkboxGroup.addThemeVariants(CheckboxGroupVariant.LUMO_VERTICAL);
+        checkboxGroup.addValueChangeListener(
+                checkboxGroupSetComponentValueChangeEvent -> {
```
Comment (tomivirkki) [inline vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java]: This event name is quite verbose "checkboxGroupSetComponentValueChangeEvent". Just "event" should be enough.

File: vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java [hunk 3]
```diff
@@ -34,174 +47,223 @@
...
-        // source-example-heading: Read-only checkbox group
-        Div valueInfo = new Div();
+        // source-example-heading: Indeterminate checkbox
+        Checkbox checkbox = new Checkbox("Select all");
+        checkbox.setIndeterminate(true);
+        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
+        Set<String> items = new LinkedHashSet<>(
+                Arrays.asList("Option one", "Option two"));
+        checkboxGroup.setItems(items);
+        List<CheckboxGroup> checkboxList = new ArrayList<>();
```
Comment (tomivirkki) [inline vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java]: This list isn't used anywhere

File: vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java (hunk 1 above)
Comment (shadikhani) [reply vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java]: Done.

File: vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java (hunk 2 above)
Comment (shadikhani) [reply vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java]: Done.

File: vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java (hunk 3 above)
Comment (shadikhani) [reply vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java]: Done.

File: vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java (hunk 1 above)
Comment (tomivirkki) [reply vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java]: The demo already has a value change listener with logic to modify the indeterminate state of the "select all" checkbox where appropriate. You should not set it manually.

Just move the `checkboxGroup.setValue(Collections.singleton("Option one"));` _after_ the value change listeners are added so there's no need for manually setting the checkbox as indeterminate.

Comment (shadikhani) [reply vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java]: Done.

File: vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java [hunk 4]
```diff
@@ -34,174 +46,219 @@
...
+        Binder<Employee> binder = new Binder<>();
+
+        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
+        checkboxGroup.setLabel("Employee title");
+        checkboxGroup.setItems("Account Manager", "Designer",
+                "Marketing Manager", "Developer");
+        checkboxGroup.addThemeVariants(CheckboxGroupVariant.LUMO_VERTICAL);
+
+        binder.forField(checkboxGroup)
+                .asRequired("Please choose an employee title")
```
Comment (tomivirkki) [inline vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java]: Is this demo copied from some other component's validation example? A lot of the demo source indicates that the user is doing a single selection for an employee's "title" ("Employee title", "Please choose an employee title", "setTitle"). Even the Bean has a field "title" but it's actually a `Set<String>` and this isn't visible in the demo source.

Comment (shadikhani) [reply vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java]: Done. (I'm going to talk about this fix with Jouni)

**Pull request merged**